src/Fit/fit_via_EWs.jl

```diff
@@ -0,0 +1,507 @@
+    logg_residual = (weighted_mean(A[neutrals], A_inv_var[neutrals]) -
+                     weighted_mean(A[.!neutrals.&finitemask], A_inv_var[.!neutrals.&finitemask]))
+
+    vmic_residual = get_slope(REWs[neutrals], A[neutrals], A_inv_var[neutrals])
+    feh_residual = weighted_mean(A[finitemask], A_inv_var[finitemask]) -
+                   (params[4] + Korg.grevesse_2007_solar_abundances[Z])
+    residuals = [teff_residual, logg_residual, vmic_residual, feh_residual]
+    residuals .*= .!fix_params # zero out residuals for fixed parameters
+
+    callback(ForwardDiff.value.(params), ForwardDiff.value.(residuals), ForwardDiff.value.(A))
+    residuals
+end
+
+# called by _stellar_param_equation_residuals
+# returns (statistical_uncertainty, systematic_uncertainty)
+function _stellar_param_residual_uncertainties(params, linelist, EW, EW_err, passed_kwargs)
+    A, A_inv_var, neutrals, REWs, _ = _stellar_param_equations_precalculation(true, params,
+                                                                              linelist, EW,
+                                                                              EW_err, passed_kwargs)
+
+    # estimated total (including systematic) err in the abundances of each line
+    total_err = std(A)
+    total_ivar = ones(length(A)) * total_err^-2
+
+    stat_sigma, total_sigma = map([A_inv_var, total_ivar]) do ivar
+        sigma_mean = 1 ./ sqrt(sum(ivar))
+        teff_residual_sigma = get_slope_uncertainty([line.E_lower for line in linelist[neutrals]],
+                                                    ivar[neutrals])
+        vmic_residual_sigma = get_slope_uncertainty(REWs[neutrals], ivar[neutrals])
+        [teff_residual_sigma, sigma_mean, vmic_residual_sigma, sigma_mean]
+    end
+
+    if EW_err == ones(length(EW))
+        # in the case that EW uncertainties were specified, the residuals were calculated
+        # assuming errs = 1, but we want to ignore them here, and call all error "systematic"
+        [0.0, 0.0, 0.0, 0.0], total_sigma
+    else
+        stat_sigma, sqrt.(max.(total_sigma .^ 2 .- stat_sigma .^ 2, 0))
+    end
+end
+
+function _stellar_param_equations_precalculation(exact_calculation, params, linelist, EW, EW_err,
+                                                 passed_kwargs)
+    teff, logg, vmic, m_H = params
+    A_X = Korg.format_A_X(m_H)
+    atm = Korg.interpolate_marcs(teff, logg, A_X; perturb_at_grid_values=true,
+                                 clamp_abundances=true)
+
+    if exact_calculation
+        A = ews_to_abundances(atm, linelist, A_X, EW; vmic=vmic, blend_warn_threshold=Inf,
+                              passed_kwargs...)
+    else
+        A = ews_to_abundances_approx(atm, linelist, A_X, EW; vmic=vmic, blend_warn_threshold=Inf,
+                                     passed_kwargs...)
+    end
+    # convert error in EW to inverse variance in A (assuming linear part of C.O.G.)
+    A_inv_var = (EW .* A ./ EW_err) .^ 2
```

Comment:
This is wrong.  But probably not so much so that it should block this PR.  I think it can be fairly safely dealt with afterwards.